arm-trusted-firmware-mvebu: enable per default on buildbot

dhewg opened the pull request:
The firmware for these boards is not part of the built image, it needs to be updated independently using a binary file and flashed to NOR using the u-boot 'bubt' command.
    
Let buildbot always build firmware binaries for all boards, so that they can be found and downloaded from the "Supplementary Files" section.

See also #3360.

dhewg committed:
uboot-mvebu: don't default to enable a3700 builds

u-boot binaries for this SoC are only required for the atf package,
disable them per default so they don't get build unnecessarily.

Signed-off-by: Andre Heider <a.heider@gmail.com>

dhewg committed:
arm-trusted-firmware-mvebu: enable per default on buildbot

The firmware for these boards is not part of the built image, it needs to
be updated independently using a binary file and flashed to NOR using the
u-boot 'bubt' command.

Let buildbot always build firmware binaries for all boards, so that they can
be found and downloaded from the "Supplementary Files" section.

Signed-off-by: Andre Heider <a.heider@gmail.com>

dhewg commented:
cc @tmn505 @hauke 
The atf package already installs to `$(BIN_DIR)/flash-image-$(BUILD_VARIANT)/`, I hope this is enough so that binaries pop up [here](https://downloads.openwrt.org/snapshots/targets/mvebu/cortexa53/) (u-boot does at least)?

erdoukki commented:
@dhewg thnaks

tmn505 commented on package/boot/uboot-mvebu/Makefile:
```diff
@@ -41,6 +41,7 @@ define U-Boot/espressobin
   BUILD_DEVICES:=globalscale_espressobin globalscale_espressobin-v7
   BUILD_SUBTARGET:=cortexa53
   UBOOT_CONFIG:=mvebu_espressobin-88f3720
+  DEFAULT:=n
```
There's no point in disabling it here, this package will be selected by atf package and incomplete u-boot will still be present in supplementary files. The u-boot.mk has default target which does that. I think enabling/disabling this final installation target could be dependent on subtarget. This will look ugly but it'll work.

dhewg replied:
The intention wasn't related to buildbot, just overwriting the default:
u-boot.mk currently defaults to enable u-boot builds for the selected board. So if you choose to build an OpenWRT image for espressobin, this is currently enabled. This overwrites it, so if someone builds for this device, u-boot for it isn't unnecessary enabled. Unless that someone chooses atf too of course.

With the other patch it will always be build on buildbot, so it wouldn't have an effect there.

tmn505 replied:
Ok, but useless u-boot.bin will still be present there which could be confusing.

dhewg replied:
Right, I added a patch to disable installing u-boot.
I discovered that it's using the nice BuildTarget/bin common code for the installation, so I added another patch so the atf package uses it too.

dhewg committed:
uboot-mvebu: don't install 64bit binaries

u-boot binaries are not useful for these boards, they need to be combined
with atf for a proper firmware.

Signed-off-by: Andre Heider <a.heider@gmail.com>

dhewg committed:
arm-trusted-firmware-mvebu: convert to PKG_TARGETS:=bin

Reuse the common BuildTarget/bin target to properly install/clean the
binaries to/from $BIN_DIR.

Signed-off-by: Andre Heider <a.heider@gmail.com>

dhewg commented:
@tmn505 do the patches look okay to you now?

tmn505 commented on package/boot/arm-trusted-firmware-mvebu/Makefile:
```diff
@@ -21,13 +21,21 @@ PKG_LICENSE_FILES:=docs/license.rst
   CATEGORY:=Boot Loaders
   DEPENDS:=@TARGET_mvebu_cortexa53
   VARIANT:=$(subst arm-trusted-firmware-mvebu-,,$(1))
+  DEFAULT:=y if BUILDBOT
```
Small nit. If any of the boards supported by OpenWrt has available bootloader, it's always built by default, so I don't think distinguishing between buildbot and own build is necessary.

dhewg replied:
That's for u-boot and what BUILD_DEVICES does (which the patch `uboot-mvebu: don't default to enable a3700 builds` disables). atf never defaulted to y afaict, else we would already see published binaries.

dhewg committed:
arm-trusted-firmware-mvebu: fix topology for ESPRESSObin V3-V5 (1GB 1CS)

Signed-off-by: Andre Heider <a.heider@gmail.com>

dhewg commented:
Added a related patch to fix one topology

hauke commented:
I applied the first 3 patches from this series to master.
Instead of always activating the build of the ATF when run by build bot, I would prefer to build it when the image is build. I ported the code which is used to build the board specific U-Boot binaries to ATF, see here:
https://git.openwrt.org/?p=openwrt/staging/hauke.git;a=shortlog;h=refs/heads/mvebu

Currently the UART in Linux does not allow RX when I use this mainline U-Boot on my espressobin. The boot cmd is wrong:.
mainline U-Boot, broken:
[    0.000000] Kernel command line: root=PARTUUID=51f0b992-02 rw rootwait
vendor U-Boot, working:
[    0.000000] Kernel command line: console=ttyMV0,115200 root=/dev/mmcblk0p2 rw rootwait

hauke commented:
I missed the changes from #3405, now it looks like this and works:
[    0.000000] Kernel command line: root=PARTUUID=10a2a153-02 rw rootwait console=ttyMV0,115200 earlycon=ar3700_uart,0xd0012000

dhewg commented:
Nice, looks great at first glance! I'll give it a spin later.

`console=` isn't part of u-boot due to a debian bug, see https://patchwork.ozlabs.org/project/uboot/patch/20200824142502.7281-4-pali@kernel.org/
But #3405 takes care of that, so it works everywhere.

dhewg commented:
@hauke working nicely for me, thanks! (nit: typo in the commit msg: `u-bot`)

pali commented:
@dhewg: was this pull request merged or is there something waiting in queue?

dhewg commented:
3 patches have been merged, the remaining 2 are superseeded by http://patchwork.ozlabs.org/project/openwrt/patch/20201011184121.14056-1-hauke@hauke-m.de/
I don't know why it's not yet merged...

hauke commented:
I pushed this change:
https://patchwork.ozlabs.org/project/openwrt/patch/20201011184121.14056-1-hauke@hauke-m.de/

This pull request can be closed then.

dhewg commented:
Nice, thanks!

It looks like it worked, binaries popped up at https://downloads.openwrt.org/snapshots/targets/mvebu/cortexa53/ ;)

pali commented:
> https://downloads.openwrt.org/snapshots/targets/mvebu/cortexa53/

Seems that this link contains wrong images or rather wrongly named images. In directory `trusted-firmware-a-espressobin-` is **secure** firmware (with concatenation of trusted firmware and untrusted u-boot image).

So rather call it just firmware... At least it is not (arm) trusted firmware.

erdoukki commented:
great that firmwares are now available !
Thanks...

pali commented:
@dhewg, @tmn505, @hauke: Could you rename directory from `trusted-firmware-a-...` to just `firmware-...`?

pali commented:
@dhewg, @tmn505, @hauke: ping

dhewg commented:
I'll have a look

dhewg commented:
The name comes from the package name in addition with its variants. Two common helpers are used for that, `include/trusted-firmware-a.mk` and `include/package-bin.mk`.
I'm not seeing a nice way to fix it (yet), and it's probably not worth the hassle for just a directory name...

pali commented:
There are `define Package/trusted-firmware-a-$(1)/install` and `Package/trusted-firmware-a/install = $(Package/trusted-firmware-a/install/default)`. Maybe some modification of these variables could help?
